packages/recorder/src/poller.ts

```diff
@@ -0,0 +1,96 @@
+import child_process from 'child_process';
+import EventEmitter from 'events';
+
+function isTaskRunning(taskMatcher: RegExp) {
+  const promise = new Promise<boolean>((resolve, reject) => {
+    child_process.exec('tasklist', function (err, stdout, _stderr) {
+      if (err) {
+        reject(err);
+      }
+      if (taskMatcher.test(stdout)) {
+        resolve(true);
+      } else {
+        resolve(false);
+      }
+      // stdout is a string containing the output of the command.
+    });
+  });
+  return promise;
+}
+
+/**
+ * The Poller singleton periodically checks the list of WoW active processes.
+ * If the state changes, it emits either a 'wowProcessStart' or
+ * 'wowProcessStop' event.
+ */
+export default class Poller extends EventEmitter {
+  private _isWowRunning = false;
+
+  private _pollInterval: ReturnType<typeof setInterval> | undefined;
+
+  private processRegex = new RegExp(/(wow(T|B|classic)?)\.exe/, 'i');
```

Comment:
Move regex out to constructor probably... this is one of the last few spots WoW specific code is living in /recorder